src/TLuaInterpreter.cpp

```diff
@@ -12015,6 +12015,74 @@ int TLuaInterpreter::sendIrc( lua_State * L )
     return 0;
 }
 
+int TLuaInterpreter::setServerEncoding(lua_State * L)
+{
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if (! pHost) {
+        lua_pushstring(L, "setServerEncoding: NULL Host pointer - something is wrong!");
+        return lua_error(L);
+    }
+
+    QString newEncoding;
+    if (! lua_isstring(L, 1)) {
+        lua_pushfstring(L, "setServerEncoding: bad argument #1 type (newEncoding as string expected, got %s!)",
+                        luaL_typename(L, 1));
+        return lua_error( L );
+    }
+    else {
+        newEncoding = QString::fromUtf8(lua_tostring(L,1));
+    }
+
+    QPair<bool, QString> results = pHost->mTelnet.setEncoding(newEncoding);
+
+    if(results.first) {
+        lua_pushboolean(L, true);
```

Comment:
The user could change the encoding themselves using the GUI, OR a script could change it without them being aware of it.  If it gets changed incorrectly it can harm the connection to the MUD Server and if we or someone else is called in to debug a user's setup we will want to know if the encoding has been changed mid stream.  It is presented as an "[ OK ] - " message and is intended to confirm something important - after all we bother to report being connected or disconnected from the MUD server and this is almost as important - for instance: entering a user name or password could depend on this setting being correct!

I repeat it will only show when the encoding is __changed__ so once it is set correctly by a user they may never see it again on that profile.